corehq/apps/events/views.py

```diff
@@ -35,6 +36,7 @@
     close_mobile_worker_attendee_cases,
     sync_mobile_worker_attendees,
 )
+from rest_framework import status
```

Comment:
I feel a bit weird about this.

On the one hand, I appreciate the instinct to use constants for values that are essentially constants.

On the other hand,
* it feels wrong to introduce a dependency on Django REST Framework when this module is not a REST API
* there are 139 instances in the codebase where we use HTTP status code 400 without using a constant, and up until now 0 instances where we do use a constant
* Django just uses integers
* the Requests library defines `requests.codes`, which feels like a better dependency
* the standard library's http module defines `http.HTTPStatus`, which feels like a much better dependency
* it would be worrying if any web developer at Dimagi does not know what HTTP status 400 means

So I don't think this is necessary, and I feel like it would be better to be consistent with the rest of the codebase than to introduce a constant this one time. But, if you feel strongly, maybe,
```suggestion
from http import HTTPStatus
```
and then use `HTTPStatus.BAD_REQUEST`.

Reply:
I do like the use of constants, for the reason that it helps to make the code more readable. This is more so for the more uncommon status codes that developers might not be too familar with and they can immediately see what it represents without having to do an online search for it. In saying that however, I have to agree that it is probably better to follow the codebase convention and not have inconsistencies in how we

Reply:
I like the use of constants as I feel like they help make code more readable. Especially for the more uncommon status codes that devs might not be too familiar with, it can be nice to immediately see what a status code represents instead of having to search for it online. Furthermore, it can make things less error prone. A developer might accidently give a status code of 403, when they meant to use 404 for example.

With the above said however, it is largely personal preference and I do agree it would be better to rather keep the codebase consistent and follow the conventions used throughout.

Reply:
> Especially for the more uncommon status codes

Yeah, I agree. I can never remember whether 301 or 302 is the "nothing to see here" redirect or the "thanks for the submission, now go here" redirect.

> and follow the conventions

And if those conventions are not perfect, we can maybe find some time to fix them, and make a new convention.
```
$ git grep 'status=[0-9]\{3\}' | wc -l
167
$ git grep 'status == [0-9]\{3\}' | wc -l
4
```
Maybe next time I find myself with nothing to do on a Friday afternoon? :upside_down_face:

Reply:
I would certainly be curious to hear how some of the other devs feel about it as well. If enough feel that the current convention could use improving, then I agree it might be something worthwhile to do at a later point if there's spare time available. :thinking:

Reply:
@zandre-eng  I'm certainly pro making a codebase-wide update to use constants, but agree with @kaapstorm that it should rely on `http` and not `rest_framework`.

Reply:
What I like about `HTTPStatus` is the fact that you're almost spelling out what you expect, instead of the geeky "403" for instance. I think it improves readability quite a lot.